File: 2do_DAM/PMDM/1ra_Evaluacion/Examen/MartinSonsecaEduardo/app/src/main/java/com/example/martinsonsecaeduardo/MainActivity.java
package com.example.martinsonsecaeduardo;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.ListView;
import android.widget.Spinner;

import com.example.martinsonsecaeduardo.databinding.ActivityMainBinding;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {
    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
        return super.onOptionsItemSelected(item);
    }

    

    private ActivityMainBinding binding;
    ArrayList<String> datosMostrar = new ArrayList<>();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        // setContentView(R.layout.activity_main);

        binding = ActivityMainBinding.inflate(getLayoutInflater());

        setContentView(binding.getRoot());
        aniadorDatos();

        Alumno alumno = new Alumno("Eduardo", "5");
        binding.setAlumno(alumno);
    }

    private void aniadorDatos() {

        ListView lvDatos = binding.lvDatos;

        lvDatos.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> adapterView, View view, int i, long l) {
                ArrayAdapter<String> adapter = new ArrayAdapter<>(getApplicationContext(), android.R.layout.simple_list_item_1, datosMostrar);

                lvDatos.setAdapter(adapter);
                String nombre = binding.etNombre.getText().toString();
                String calificacion = (String) binding.spCalificacion.getSelectedItem();
                Alumno alumno = new Alumno(nombre, calificacion);
                datosMostrar.add(alumno.toString());

                Log.d(getAttributionTag(), "Datos" + "Los datos son " + alumno.toString());


                adapter.notifyDataSetChanged(); //Para que no e vaya "Pisando" el dato
            }
        });
    }
}

File: 2do_DAM/PMDM/1ra_Evaluacion/Ejemplos/Kotlin/Ejemplo014_RecycleViewCardView/app/src/main/java/com/example/ejem14recycleviewcardview/MainActivity.kt
package com.example.ejem14recycleviewcardview

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.constraintlayout.widget.ConstraintLayout
import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.example.ejem14recycleviewcardview.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    lateinit var binding:ActivityMainBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        //binding.rvListado.layoutManager = LinearLayoutManager(this)
        binding.rvListado.layoutManager = LinearLayoutManager(this, LinearLayoutManager.HORIZONTAL, false)

        //binding.rvListado.layoutManager = GridLayoutManager(this,1)
        binding.rvListado.adapter = PersonasAdapter(PersonasProvider.datos)
    }


}